packages/block-library/src/navigation/view.js

```diff
@@ -63,11 +61,13 @@ window.addEventListener( 'load', () => {
 			'.wp-block-navigation-item.has-child'
 		);
 		submenuBlocks.forEach( function ( block ) {
-			if (
-				! block.contains( event.target ) ||
-				( block.contains( event.target ) && event.key === 'Escape' )
-			) {
+			if ( ! block.contains( event.target ) ) {
+				closeSubmenus( block );
+			} else if ( event.key === 'Escape' ) {
+				const toggle = block.querySelector( '[aria-expanded="true"]' );
 				closeSubmenus( block );
+				// Focus the submenu trigger so focus does not get trapped in the closed submenu.
+				toggle.focus();
```

Comment:
When testing with a menu with nested submenu, this triggers an error:

`Uncaught TypeError: Cannot read properties of null (reading 'focus')`

That's because:
- the script loops through all the submenu blocks
- for each submenu block, it gets the toggle with an aria-expanded="true" attribute
- then it closes all the submenu and switches aria-expanded to false
- at this point, it loops to the second submenu block, where the toggle has now an aria-expanded set to false
- querySelector returns null and `toggle` is null
- attempting to set focus on `null` triggers the error

I think it would be sufficient to check for the toggle existence before attempting to set focus on it or alternatively return it it's null before setting focus.

Reply:
@afercia Better now?

Reply:
Looks good to me!